src/journal/test-catalog.c

```diff
@@ -32,9 +33,8 @@ static const char *no_catalog_dirs[] = {
         NULL
 };
 
-static Hashmap * test_import(const char* contents, ssize_t size, int code) {
-        int r;
-        char name[] = "/tmp/test-catalog.XXXXXX";
+static Hashmap* test_import(const char* contents, ssize_t size, int code) {
+        _cleanup_(unlink_tempfilep) char name[] = "/tmp/test-catalog.XXXXXX";
```

Comment:
hmm, so for newer code I usually used this:

```c
_cleanup_(unlink_and_freep) char *tmp = NULL;

r = tmpfn_random_child(…, &tmp);
```
 or so... which has the nice little benefit that it doesn't create the ambuigity that XXXXX actually is a valid suffix that mkostemp() and friends  might output, in an exceptional corner case.

Also, there's also rmdir_and_free() and rm_rf_physical_and_free() already, which makes this even more powerful to also cover directories and suchlike